### ISSUE TITLE
How to initialize model from checkpoint correctly and load all pre-trained weights?

### ISSUE THREAD

**Issue by eugenegoncharov** [tensorflow/tpu#328]
I'm using deeplab model with pretrained weights.
When running on GPU the initial loss is somewhere around 1.3
When loading the same pretrained weights on TPU the loss is 4 times larger.

It seems like not all layer weights are loaded. 

Can't find what causing the issue. Double checked all the initialization code and tried couple of different ways to preload the pretrained weights into the model

**Comment by saberkun** [comment]
On TPU, do you use scaffold_fn to load using init_from_checkpoint? See example: https://github.com/google-research/bert/blob/master/run_classifier.py#L656

**Comment by eugenegoncharov** [comment]
Yes, I'm using the scaffold_fn the same way as in the example.

I've tried to use scaffold_fn with the init_fn as well. Both ways seem to initialize the checkpoint somehow, but not the same way as on GPU/CPU.

**Comment by eugenegoncharov** [comment]
I'm using TensorFlow 1.13 and TPU v3-8

**Comment by WonderAndMaps** [comment]
Hi, you can try `warm_start_from` in https://www.tensorflow.org/api_docs/python/tf/contrib/tpu/TPUEstimator#__init__
